semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -1361,29 +1550,41 @@ and map_function_value_parameters (env : env)
 
 and map_guard_statement (env : env) ((v1, v2, v3, v4, v5) : CST.guard_statement)
     =
+  (* A guard statement seems to be semantically equivalent to an "if not".
+   *)
   let v1 = (* "guard" *) token env v1 in
   let v2 = map_if_condition_sequence_item env v2 in
   let v3 =
     Common.map
       (fun (v1, v2) ->
         let v1 = (* "," *) token env v1 in
         let v2 = map_if_condition_sequence_item env v2 in
-        todo env (v1, v2))
+        v2)
       v3
   in
-  let v4 = (* else *) token env v4 in
-  let v5 = map_function_body env v5 in
-  todo env (v1, v2, v3, v4, v5)
+  let cond = combine_conds v2 v3 in
+  let body = map_function_body env v5 in
+  G.If (v1, G.Cond (G.special (G.Op G.Not, G.fake "not") [ cond ]), body, None)
+  |> G.s
+
+and combine_conds cond conds =
+  List.fold_left
+    (fun acc x ->
+      G.Call
+        ( G.IdSpecial (G.Op G.And, G.fake "&&") |> G.e,
+          G.fake_bracket [ G.Arg acc; G.Arg x ] )
+      |> G.e)
+    cond conds
```

Comment:
We've discussed this separately, but could you add a comment here justifying this desugaring?